sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/BooleanLiteralUnnecessary.cs

```diff
@@ -264,6 +241,14 @@ private static bool IsNullableBoolean(ITypeSymbol type)
             return false;
         }
 
+        private static bool ShouldNotReport(ITypeSymbol typeLeft, ITypeSymbol typeRight)
+        {
+            return typeLeft == null
+                || typeRight == null
+                || IsNullableBoolean(typeLeft)
+                || IsNullableBoolean(typeRight);
+        }

```

Comment:
We should not report if one of the type symbols is unknown (e.g. ErrorType). The actual problem is that the type symbol of the argument in `All(x => x.IsStateValid == true)` was unknown and we raised a FP.